pyccel/ast/numpyext.py

```diff
@@ -333,7 +333,7 @@ def __new__(cls, arg, index=None):
                                 IndexedBase)):
             raise TypeError('Uknown type of  %s.' % type(arg))
 
-        # TODO add check on index: must be Integer or Variable with dtype=int
+        # TODO add check on index: must be sp_Integer or Variable with dtype=int
         # TODO [YG, 09.10.2018]: Verify why index should be passed at all (not in Numpy!)
```

Comment:
I actually meant the comment below, which dates back to October 2018! :wink:

Reply:
I got rid of both as I don't think we need to ensure that the index is an Integer or Variable with dtype=int as this will pose problems fo rthings such as x[i*j] where the type of index would be Mul. If every object has a dtype then it may be possible to test this